R/CriteriaInputs.R

```diff
@@ -0,0 +1,699 @@
+#' the paramRef data frame which contains an additional column name: 'organization_name'
+#' in order to determine the proper crosswalk between TADA.ComparableDataIdentifier and ATTAINS.ParameterName
+#' by organization name.
+#'
+#' @return A data frame which contains the columns: TADA.ComparableDataIdentifier, organization_name,
+#' EPA304A.PollutantName, ATTAINS.ParameterName, and ATTAINS.FlagUseName. Users will need to review
+#' the crosswalk between ATTAINS.ParameterName, use_name and TADA.ComparableDataIdentifier.
+#'
+#' @export
+#'
+#' @examples
+#' # Users are required to provide a paramRef argument first. See TADA_CreateParamRef() for additional comments.
+#' paramRef_UT <- TADA_CreateParamRef(Data_Nutrients_UT, org_names = "Utah", excel = FALSE)
+#'
+#' paramRef_UT2 <- dplyr::mutate(paramRef_UT, ATTAINS.ParameterName = dplyr::case_when(
+#'   TADA.CharacteristicName == "AMMONIA" ~ "AMMONIA, TOTAL",
+#'   TADA.CharacteristicName == "NITRATE" ~ "NITRATE",
+#'   TADA.CharacteristicName == "NITROGEN" ~ "NITRATE/NITRITE (NITRITE + NITRATE AS N)"
+#' ))
+#' paramRef_UT3 <- TADA_CreateParamRef(Data_Nutrients_UT, paramRef = paramRef_UT2, org_names = "Utah", excel = FALSE)
+#'
+#' # After running TADA_CreateParamRef() users will provide this as a function input. First example is only for the org Utah
+#' paramUseRef_UT <- TADA_CreateParamUseRef(Data_Nutrients_UT, paramRef = paramRef_UT3, org_names = c("Utah"), excel = FALSE)
+#'
+#' # Users can include the EPA304a standards by itself or compared to their org(s)
+#' paramUseRef_UT2 <- TADA_CreateParamUseRef(Data_Nutrients_UT, paramRef = paramRef_UT3, org_names = c("EPA304a", "Utah"), excel = FALSE)
+#' paramUseRef_UT3 <- TADA_CreateParamUseRef(Data_Nutrients_UT, paramRef = paramRef_UT3, org_names = c("EPA304a"), excel = FALSE)
+#' 
+TADA_CreateParamUseRef <- function(.data, org_names = NULL, paramRef = NULL, excel = FALSE, overwrite = FALSE) {
+  # overwrite argument should only be used when creating an excel file.
+  if (excel == FALSE && overwrite == TRUE) {
+    stop("argument input excel = FALSE and overwrite = TRUE is an invalid combination. Cannot overwrite the excel generated spreadsheet if a user specifies excel = FALSE")
+  }
+  
+  # Checks if ref contains a dataframe and necessary columns to proceed.
+  if (is.null(paramRef)) {
+    paramRef <- openxlsx::read.xlsx(downloads_path, sheet = "CreateParamRef")
+  }
+  
+  if (is.null(org_names)) {
+    print("No organization name provided, users should provide a list of ATTAINS domain organization state or tribal name that pertains to their dataframe. Attempting to pull in organization names found in the TADA data frame.")
+  }
+  
+  if (sum(!is.na(paramRef$ATTAINS.ParameterName)) == 0) {
+    stop("No values were found in ATTAINS.ParameterName. Please ensure that you have inputted all field values of interest in the ATTAINS.ParameterName column generated from TADA_CreateParamRef() function")
```

Comment:
I might want to change this to a warning. I am considering the case in which a user is only interested in the EPA304a standards. In this scenario, there is not a need for a user to provide a crosswalk between ATTAINS.ParameterName and TADA.ComparableDataIdentifier as it will only leverage the EPA304aPollutantName.